hips/hip-9999.md

```diff
@@ -0,0 +1,90 @@
+---
+hip: "9999"
+title: "Add export-values directive (similar to import-values)"
+authors: [ "Jan von Loewenstein <jan.von.loewenstein@sap.com>", "Johannes Dillmann <j.dillmann@sap.com>", "Pavel Busko <pavel.busko@sap.com>", "Philipp Stehle <philipp.stehle@sap.com>", "Ralf Pannemans <ralf.pannemans@sap.com>", "Sumit Kulhadia <sumit.kulhadia@sap.com>" ]
+created: "2022-02-01"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+
+Composing Helm charts is currently limited because values can only be distributed from a subchart to the composite chart via the [`import-values`][import-values] directive.
+
+This HIP describes the `export-values` directive which is similar to the `import-values`[import-values] directive, and allows exporting values from the composite chart to the subchart.
+
+With that the composite chart can define the values freely and is not limited to stick to the structure defined in the subchart. In some cases, this would result in a better user experience.
+
+This change was discussed in helm/#3242. There were several attempts to implement this: helm/#7477, helm/#3243.
+
+## Specification
```

Comment:
I think we should be more precise in what we are describing here. Shouldn't we show user, parent and child values here. Similarly, we should show the Chart.yaml for both parent and child charts.

Whether adding a grandchild chart adds any further value is currently not clear to me, but we have to make sure that the proposal we make is sound for any depth of nesting.

Reply:
> Shouldn't we show user, parent and child values here. 

They are rather trivial, right. But I guess it makes sense anyhow.

```yaml
port: 8080
```

> Similarly, we should show the Chart.yaml for both parent and child charts.

The `Chart.yaml` of the child charts are irrelevant, aren't they?

> Whether adding a grandchild chart adds any further value is currently not clear to me, but we have to make sure that the proposal we make is sound for any depth of nesting.

We thought about this. It would make the example more complicated. But it makes sense to think this through...

Reply:
1. We could add the other files (values.yaml, chart.yaml etc) but this is really not adding any value to the HIP proposal. 
2. We quickly tested the existing `import-values` directive and found out that it does not support the grand-child level. So in order to be aligned with the existing functionality, we decided not to include it in our proposal.